packages/core/README.md

```diff
@@ -78,117 +62,378 @@ app.all("/mcp", async (c) => {
+You can begin with a single file server and add state only when you need it. Adapters let you swap in storage or queueing code without touching tools or handlers.
 
-### Stateless Mode (Default)
-No session support, no GET endpoint for SSE streaming.
+### Scaling playbook
 
-```typescript
-import { StreamableHttpTransport } from "mcp-lite";
+- **Local prototyping:** run the transport with no adapters. Every request is stateless and there is nothing to clean up.
+- **Single server:** add `InMemorySessionAdapter` (and optionally `InMemoryClientRequestAdapter`) to keep progress events and elicitations alive across multiple requests from the same client.
+- **Distributed or serverless:** implement the adapter interfaces against Redis, KV, Durable Objects, or queues. See the Cloudflare Workers example for a working KV-backed adapter.
 
-// Stateless mode - no session management
-const transport = new StreamableHttpTransport();
-const httpHandler = transport.bind(mcp);
-```
+### Deployment Patterns
 
-### Stateful Mode with Sessions
-Enable sessions and SSE streaming by providing a `SessionAdapter`:
+| Environment | Session Storage | State Storage | Transport Configuration |
+|-------------|----------------|---------------|------------------------|
+| Development | None | N/A | `StreamableHttpTransport()` |
+| Single server | In-memory | In-memory | `InMemorySessionAdapter` |
+| Distributed | Redis/KV | Redis/KV | Custom adapters |
+| Serverless | KV/R2 | Durable Objects | See examples below |
+
+### Adapter Configuration
 
 ```typescript
-import { StreamableHttpTransport, InMemorySessionAdapter } from "mcp-lite";
+// Development: stateless
+const transport = new StreamableHttpTransport()
 
-// Stateful mode with sessions and SSE support
+// Production: with sessions and client requests
 const transport = new StreamableHttpTransport({
   sessionAdapter: new InMemorySessionAdapter({
-    maxEventBufferSize: 1024  
+    maxEventBufferSize: 1024
+  }),
+  clientRequestAdapter: new InMemoryClientRequestAdapter({
+    defaultTimeoutMs: 30000
   })
-});
+})
+```
 
-const httpHandler = transport.bind(mcp);
+### Built-in Adapters
+
+- `InMemorySessionAdapter` - Session storage in memory
+- `InMemoryClientRequestAdapter` - Client request tracking in memory
+
+### Client Requests and Elicitation
+
+The server can send JSON-RPC requests back to the MCP client (for example when you call `ctx.elicit`). Those requests are routed through the `ClientRequestAdapter`. Provide one when you need:
+- Timeouts or retries for client prompts.
+- To make sure an elicitation response is delivered even when the original POST is finished.
+- To back the pending requests with shared storage in a multi-instance deployment.
+
+The in-memory adapter covers local runs. For production you can implement `ClientRequestAdapter` using Redis, D1, Durable Objects, or any queue that can look up pending requests by session and request id.
```

Comment:
again this is very cloudflare-skewed. probably better to change "D1" to just be ... sqlite? or at that point just say "a database" idk